### PR TITLE
fix issue #231 

### PR DESCRIPTION
**Pull request by Esonhugh**
Link issue #231

It's possible a fix for #231 . But I still have no idea why _GNU_SOURCE is defined in something like Havoc.

### CODE REVIEW

**Comment by ToruNiina** [comment]
It seems that github actions does not support runner named macos-13-arm64. Although there is a doc about macos-13-arm64 (https://github.com/actions/runner-images/tree/main/images/macos), but the job does not run. I searched other repositories that tries to run it by https://github.com/search?q=macos-13-arm64&type=code. I looked 2 or 3 of those, but none of them are running.

And it seems to have not been supported yet, I think we can just comment out macos-13-arm64 section for now.
https://docs.github.com/ja/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -138,6 +138,56 @@ jobs:
...
+        run: |
+            mkdir build && cd build
+            cmake .. -Dtoml11_BUILD_TEST=ON -DCMAKE_CXX_STANDARD=${{ matrix.standard }} -DTOML11_USE_UNRELEASED_TOML_FEATURES=${{ matrix.unreleased }}
+      - name: Build
+        run: |
+            cd build && cmake --build .
+      - name: Test
+        run: |
+            cd build && ctest --output-on-failure
+
```
Comment (ToruNiina) [inline .github/workflows/main.yml:165]:
```suggestion
#  build-osx-13-arm64:
#    runs-on: macos-13-arm64
#    strategy:
#      matrix:
#        standard: ['11', '14', '17', '20']
#        unreleased: ['ON', 'OFF']
#    steps:
#      - name: Checkout
#        uses: actions/checkout@v3
#        with:
#          submodules: true
#      - name: Install
#        run: |
#            brew install boost
#      - name: Configure
#        run: |
#            mkdir build && cd build
#            cmake .. -Dtoml11_BUILD_TEST=ON -DCMAKE_CXX_STANDARD=${{ matrix.standard }} -DTOML11_USE_UNRELEASED_TOML_FEATURES=${{ matrix.unreleased }}
#      - name: Build
#        run: |
#            cd build && cmake --build .
#      - name: Test
#        run: |
#            cd build && ctest --output-on-failure

```

**Comment by Esonhugh** [comment]
> It seems that github actions does not support runner named macos-13-arm64. Although there is a doc about macos-13-arm64 (https://github.com/actions/runner-images/tree/main/images/macos), but the job does not run. I searched other repositories that tries to run it by https://github.com/search?q=macos-13-arm64&type=code. I looked 2 or 3 of those, but none of them are running.
> 
> And it seems to have not been supported yet, I think we can just comment out macos-13-arm64 section for now. https://docs.github.com/ja/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

It looks same on my side. But What's the ClusterFuzzlite. It works on my repo, but failed in PR. : ?

**Comment by ToruNiina** [comment]
> But What's the ClusterFuzzlite.

I think that error is not caused by this change.
I have no idea why it fails here instead of the PullReq of ClusterFuzzlite, but I think I have just fixed it in #234 .

The input contains over 2000 `[` and it causes stack overflow on ci docker env because of recursive call of `parse_value` and `parse_array`. It successfully returns an error message in my local env, but somehow fails in docker running on ci, maybe because of some difference in flags or something. I added a parameter to detect deep nest of array or inline table and now it can throw an error instead of stack overflow.

Hopefully, it works when we re-run ci.

**Comment by ToruNiina** [comment]
Okay it works with `macos-13`. merging.

**Comment by Esonhugh** [comment]
Great

**Pull request merged**